includes/Core/Authentication/Setup_V2.php

```diff
@@ -0,0 +1,148 @@
+use Google\Site_Kit\Core\Permissions\Permissions;
+
+/**
+ * Class for v2 authentication setup.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Setup_V2 extends Setup {
+
+	/**
+	 * Registers functionality through WordPress hooks.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function register() {
+		add_action( 'admin_action_' . Google_Proxy::ACTION_SETUP_START, array( $this, 'handle_action_setup_start' ) );
+		add_action( 'admin_action_' . Google_Proxy::ACTION_VERIFY, array( $this, 'handle_action_verify' ) );
+		add_action( 'admin_action_' . Google_Proxy::ACTION_EXCHANGE_SITE_CODE, array( $this, 'handle_action_exchange_site_code' ) );
+	}
+
+	/**
+	 * Handles the setup start action, taking the user to the proxy setup screen.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function handle_action_setup_start() {
+		$nonce        = $this->context->input()->filter( INPUT_GET, 'nonce', FILTER_SANITIZE_STRING );
+		$redirect_url = $this->context->input()->filter( INPUT_GET, 'redirect', FILTER_SANITIZE_URL );
+
+		$this->verify_nonce( $nonce, Google_Proxy::ACTION_SETUP_START );
+
+		if ( ! current_user_can( Permissions::SETUP ) ) {
+			wp_die( esc_html__( 'You have insufficient permissions to connect Site Kit.', 'google-site-kit' ) );
+		}
+
+		if ( ! $this->credentials->using_proxy() ) {
+			wp_die( esc_html__( 'Site Kit is not configured to use the authentication proxy.', 'google-site-kit' ) );
+		}
+
+		if ( $this->credentials->has() ) {
+			$this->google_proxy->sync_site_fields( $this->credentials, 'sync' );
+		}
+
+		if ( $redirect_url ) {
+			$this->user_options->set( OAuth_Client::OPTION_REDIRECT_URL, $redirect_url );
+		}
+
+		$this->redirect_to_proxy();
+	}
+
+	/**
+	 * Handles the action for verifying site ownership.
+	 *
+	 * @since n.e.x.t
+	 */
+	public function handle_action_verify() {
+		$input               = $this->context->input();
+		$step                = $input->filter( INPUT_GET, 'step', FILTER_SANITIZE_STRING );
+		$nonce               = $input->filter( INPUT_GET, 'nonce', FILTER_SANITIZE_STRING );
+		$code                = $input->filter( INPUT_GET, 'googlesitekit_code', FILTER_SANITIZE_STRING );
+		$site_code           = $input->filter( INPUT_GET, 'googlesitekit_site_code', FILTER_SANITIZE_STRING );
+		$verification_token  = $input->filter( INPUT_GET, 'googlesitekit_verification_token', FILTER_SANITIZE_STRING );
+		$verification_method = $input->filter( INPUT_GET, 'googlesitekit_verification_token_type', FILTER_SANITIZE_STRING );
+
+		$this->verify_nonce( $nonce );
+
+		if ( ! current_user_can( Permissions::SETUP ) ) {
+			wp_die( esc_html__( 'You don\'t have permissions to set up Site Kit.', 'google-site-kit' ), 403 );
+		}
+
+		if ( ! $code ) {
+			wp_die( esc_html__( 'Invalid request.', 'google-site-kit' ), 400 );
+		}
+
+		if ( ! $verification_token || ! $verification_method ) {
+			wp_die( 'Verifying site ownership requires a token and verification method.', 400 );
```

Comment:
This should probably also be wrapped in `esc_html__`?

Reply:
Oops! Done 👍